src/main/java/nl/knaw/dans/ingest/core/service/DepositIngestTask.java

```diff
@@ -221,6 +221,13 @@ void validateDeposit() {
                     "Bag was not valid according to Profile Version %s. Violations: %s",
                     result.getProfileVersion(), violations)
                 );
+            } else {
```

Comment:
Avoid negatives in if-conditions. Better would be `if(result.isCompliant) ensureSha1ManifestPresent() else ...`.

Better still is to always `ensureSha1ManifestPresent` before calling the validator, so you have a doublecheck on your (possibly added) SHA1-manifest.

Reply:
Not sure I agree. You would repeat an expensive calculation and you don't know for sure any more whether violations are about the original bag or the altered bag.